docs/source/_src_user_guide/state_propagation/propagation_setup/propagation_architecture.rst

```diff
@@ -0,0 +1,81 @@
+.. _propagation_architecture:
+
+========================
+Propagation Architecture
+========================
+
+The core of a numerical integration is the succesive evaluation of the state derivative :math:`\dot{\mathbf{x}}=\mathbf{f}(\mathbf{x},s;\mathbf{p})`. Here, :math:`\mathbf{x}` is the propagated state which may be a combination of any types of dynamics of any set of bodies (see :ref:`TODO`), and may also include the variational equations associated with this dynamics. The variable :math:`s` is the indpendent variable of the differential equation governing the state (typically but not necesarilly the time :math:`t`), and the vector :math:`\mathbf{p}` denotes a set of parameters which are held constant during the propagation, but which do influence the solution of the equations of motion.
+
+.. _propagator_pre_processing:
+
+Propagator pre-processing
+=========================
+
+Before starting the propagation, several steps are taken to initialize the variables/objects required for the propagation. Below, we provide an overview of the steps that are taken for the propagation of a single arc (typical Tudat propagation):
+
+* It is checked whether the propagator settings are feasible (e.g. no body A propagated w.r.t. B, *and* body B propagated w.r.t. body A, etc.).
+* Several objects are created that are used during/after the propagation
+  
+  * An ``EnvironmentUpdater`` object (in C++; not exposed to Python), which decided which environment models need to be updated for each function evaluation (:ref:`single_propagation_evaluation:` and how)
+  * A ``ReferenceFrameManager`` object (in C++; not *yet* exposed to Python), which allows translations between ephemerides with different origins to be performed
+  * If the ``set_integrated_result`` to the :class:`~tudatpy.numerical_simulation.SingleArcSimulator` is set to true: A set of ``IntegratedStateProcessor`` objects (in C++; not exposed to Python) which is used to post-process the propagation results, and make any required transformations to reset the ephemerides of the propagated bodies.   
+  * A ``DynamicsStateDerivativeModel`` (in C++; not exposed to Python) object that handles the calculation of a single state derivative evaluation (see :ref:`single_propagation_evaluation`)
+  * A ``PropagationTerminationCondition`` (in C++; not exposed to Python) object that checks whether the propagation is terminated for a given time/state pair
+  * A set of functions that extract the dependent variables from the environment is created (if any)
+* The initial state is converted from conventional formulation (in which it *has* to be provided) to propagated formulation (see :ref:`conventional_propagated_states`)
+
+.. _single_propagation_evaluation:
+
+A single function evaluation
```

Comment:
what about changing the title to "What happens during a single function evaluation?" or smtg along this line? I am not entirely happy with this reformulation though, but smtg that I cannot precisely point out triggers me with the current title (same for the "A single time step)